framework/encode/d3d12_capture_manager.cpp

```diff
@@ -1770,21 +1851,156 @@ void D3D12CaptureManager::PreProcess_ID3D12CommandQueue_ExecuteCommandLists(ID3D
         }
     }
 
-    PreQueueSubmit();
+    // Split commandlist is for trim drawcalls. It means that this is a extra ExecuteCommandLists. It shouldn't count
+    // queue_submit_count_.
+    if (!is_split_commandlist)
+    {
+        PreQueueSubmit();
+    }
+}
+
+void D3D12CaptureManager::OverrideID3D12CommandQueue_ExecuteCommandLists(ID3D12CommandQueue_Wrapper* wrapper,
+                                                                         UINT                        num_lists,
+                                                                         ID3D12CommandList* const*   lists)
+{
+    bool is_complete = false;
+    if (IsTrimEnabled() && GetTrimBoundary() == CaptureSettings::TrimBoundary::kDrawCalls)
+    {
+        auto trim_draw_calls = GetTrimDrawCalls();
+        // TODO: When queue_submit_count_ becomes 0-based, remove "-1".
+        if ((common_manager_->GetQueueSubmitCount() - 1) == trim_draw_calls.submit_index)
+        {
+            auto                            target_cmdlist = lists[trim_draw_calls.command_index];
+            ID3D12CommandList_Wrapper*      target_wrapper = nullptr;
+            target_cmdlist->QueryInterface(IID_IUnknown_Wrapper, reinterpret_cast<void**>(&target_wrapper));
+            GFXRECON_ASSERT(target_wrapper);
+
+            auto target_info = target_wrapper->GetObjectInfo();
+            GFXRECON_ASSERT(target_info);
+
+            if (!target_info->is_split_commandlist)
+            {
+                DecrementCallScope();
+                // target of splitted
+                common_manager_->ActivateTrimmingDrawCalls(format::ApiFamilyId::ApiFamily_D3D12);
+
+                std::vector<ID3D12CommandList*> cmdlists;
+
+                auto target_draw_call_cmd = target_info->split_command_sets[graphics::dx12::kDrawCallArrayIndex].list;
+                GFXRECON_ASSERT(target_draw_call_cmd);
+                cmdlists.emplace_back(target_draw_call_cmd);
+
+                wrapper->ExecuteCommandLists(cmdlists.size(), cmdlists.data());
```

Comment:
1910 is also changed to `wrapper`. Since I have to change the native queue to wrapper in `PreProcess_ID3D12CommandQueue_ExecuteCommandLists` to track resource states. For consistency, I also changed here and 1910.  But it's ok to use native. I just needs to add `Encode_ID3D12CommandQueue_ExecuteCommandLists` to write it. Do you prefer using native queue?

Reply:
I see. I think it would be good to avoid the need to call `Decrement/IncrementScope` so if switching to use native handle isn't too messy that would be my preference.

Reply:
I just added a commit that moved the extra `ExecuteCommandLists` from `PreProcess_ID3D12CommandQueue_ExecuteCommandLists` to `OverrideID3D12CommandQueue_ExecuteCommandLists`, so all three extra `ExecuteCommandLists` are in `OverrideID3D12CommandQueue_ExecuteCommandLists` to make the code simpler.

In this case, `OverrideID3D12CommandQueue_ExecuteCommandLists` has to call `Decrement/IncrementScope`  since the first extra `ExecuteCommandLists` needs to be tracked.  I could limit `Decrement/IncrementScope` only for the first `ExecuteCommandLists`. The second and the third use native queue if that is you preference.

Reply:
I update that commit that only the first `ExecuteCommandLists` use the wrapped queue. `Decrement/IncrementScope` is only limited on the first `ExecuteCommandLists`. The rest of two use the native queue.